examples/ai-core/src/test/anthropic-web-search-server-side.ts

```diff
@@ -0,0 +1,60 @@
+import 'dotenv/config';
+import { anthropic } from '@ai-sdk/anthropic';
+import { generateText, streamText } from 'ai';
+
+async function testGenerateText() {
+  const result = await generateText({
+    model: anthropic('claude-3-5-sonnet-latest'),
+    prompt:
+      'What are the latest breakthroughs in quantum computing? Please search for recent developments.',
+    providerOptions: {
+      anthropic: {
+        webSearch: {
+          maxUses: 3,
+          allowedDomains: ['arxiv.org', 'nature.com', 'mit.edu'],
+        },
+      },
+    },
```

Comment:
wouldnt this need to be a tool (vs provider options), also below and other examples